auth/saml/request.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2015 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package saml
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/tsuru/config"
+	"github.com/tsuru/tsuru/db"
+	"github.com/tsuru/tsuru/errors"
+	"github.com/tsuru/tsuru/log"
+	"gopkg.in/mgo.v2/bson"
+)
+
+var (
+	requestExpire            time.Duration
+	defaultRequestExpiration = 3 * 60 * time.Second
+	ErrRequestNotFound       = &errors.ValidationError{Message: "request not found or expired"}
+)
+
+type Request struct {
+	ID       string    `json:"id"`
+	Creation time.Time `json:"creation"`
+	Expires  time.Time `json:"expires"`
+	Email    string    `json:"email"`
+	Authed   bool      `json:"authed"`
+}
+
+func loadExpireTime() error {
```

Comment:
This function will never returns an error. I believe that a better approach is remove the global vars `requestExpire` and `defaultRequestExpiration` and change this function to returns the expire time instead to load it in the global vars.